### PR TITLE
chore: add permission boundary policy name to progress tracker if present

### PR DESCRIPTION
**Pull request by huanjani**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/template/templates/app/app.yml [hunk 1]
```diff
@@ -30,7 +30,11 @@ Conditions:
 Resources:
   AdministrationRole:
     Metadata:
+      {{- if .PermissionsBoundary}}
+      'aws:copilot:description': 'A StackSet admin role with permissions boundary {{.PermissionsBoundary}} assumed by CloudFormation to manage regional stacks'
```
Comment (paragbhingre) [inline internal/pkg/template/templates/app/app.yml]: Just a one-liner alternative as there are many places where similar code is there. 
```suggestion
      'aws:copilot:description':  {{ if .PermissionsBoundary}} 'A StackSet admin role with permissions boundary {{.PermissionsBoundary}} assumed by CloudFormation to manage regional stacks' {{ else}} 'A StackSet admin role assumed by CloudFormation to manage regional stacks' {{ end }}
```

**Review by paragbhingre — COMMENTED** [review]
Looks good, just one alternative.

File: internal/pkg/template/templates/environment/partials/cfn-execution-role.yml [hunk 2]
```diff
@@ -2,7 +2,11 @@
 # See #1533.
 CloudformationExecutionRole:
   Metadata:
+    {{- if .PermissionsBoundary}}
+    'aws:copilot:description': 'An IAM Role with permissions boundary {{.PermissionsBoundary}} for AWS CloudFormation to manage resources'
+    {{- else}}
```
Comment (Lou1415926) [inline internal/pkg/template/templates/environment/partials/cfn-execution-role.yml]: nit: alternatively

```suggestion
    'aws:copilot:description': 'An IAM Role {{if .PermissionsBoundary -}} with permissions boundary {{.PermissionsBoundary}} {{- end}} for AWS CloudFormation to manage resources'
```

**Review by Lou1415926 — APPROVED** [review]
DNM label in case you want to address!

File: internal/pkg/template/templates/app/app.yml (hunk 1 above)
Comment (iamhopaul123) [reply internal/pkg/template/templates/app/app.yml]: Can we possibly make it a function that takes a string and output a processed string when `{{ if .PermissionsBoundary}}`? The benefit is that we don't need to repeat ourselves whenever we add new roles or want to modify the description.

Comment (huanjani) [reply internal/pkg/template/templates/app/app.yml]: Yeah, I'll think about it and possibly do it in a different PR!

File: internal/pkg/template/templates/app/app.yml [hunk 3]
```diff
@@ -30,7 +30,7 @@ Conditions:
 Resources:
   AdministrationRole:
     Metadata:
-      'aws:copilot:description': 'A StackSet admin role assumed by CloudFormation to manage regional stacks'
+      'aws:copilot:description': 'A StackSet admin role {{- if .PermissionsBoundary}} with permissions boundary {{.PermissionsBoundary}} {{- end}} assumed by CloudFormation to manage regional stacks'
```
Comment (iamhopaul123) [inline internal/pkg/template/templates/app/app.yml:33]: This seems to repeat a lot...and also for upcoming new roles we need this as well. Maybe consider making it a template function?

Comment (iamhopaul123) [reply internal/pkg/template/templates/app/app.yml:33]: oh nvm...

**Review by iamhopaul123 — APPROVED** [review]

**Pull request merged**